process/block/interceptedBlocks/interceptedMetaBlockHeader.go

```diff
@@ -0,0 +1,116 @@
+package interceptedBlocks
+
+import (
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+// InterceptedMetaHeader represents the wrapper over the meta block header struct
+type InterceptedMetaHeader struct {
+	hdr                 *block.MetaBlock
+	marshalizer         marshal.Marshalizer
+	hasher              hashing.Hasher
+	multiSigVerifier    crypto.MultiSigVerifier
+	chronologyValidator process.ChronologyValidator
+	shardCoordinator    sharding.Coordinator
+	hash                []byte
+}
+
+// NewInterceptedMetaHeader creates a new instance of InterceptedMetaHeader struct
+func NewInterceptedMetaHeader(arg *ArgInterceptedBlockHeader) (*InterceptedMetaHeader, error) {
+	err := checkArgument(arg)
+	if err != nil {
+		return nil, err
+	}
+
+	hdr := &block.MetaBlock{
+		ShardInfo: make([]block.ShardData, 0),
+		PeerInfo:  make([]block.PeerData, 0),
+	}
+	err = arg.Marshalizer.Unmarshal(hdr, arg.HdrBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	inHdr := &InterceptedMetaHeader{
+		hdr:                 hdr,
+		marshalizer:         arg.Marshalizer,
+		hasher:              arg.Hasher,
+		multiSigVerifier:    arg.MultiSigVerifier,
+		chronologyValidator: arg.ChronologyValidator,
+		shardCoordinator:    arg.ShardCoordinator,
+	}
+	inHdr.processFields(arg.HdrBuff)
```

Comment:
Maybe you can store in arg.HrdBuff hashed HdrBuff and remove processFields from constructor method.